kernel_tuner/hyper.py

```diff
@@ -32,58 +40,59 @@ def tune_hyper_params(target_strategy, hyper_params, *args, **kwargs):
+    cachefile = get_random_unique_filename('temp_', '.json')
+    kwargs['cache'] = str(cachefile)
 
     def put_if_not_present(target_dict, key, value):
         target_dict[key] = value if key not in target_dict else target_dict[key]
 
-    put_if_not_present(kwargs, "verbose", False)
-    put_if_not_present(kwargs, "quiet", True)
-    put_if_not_present(kwargs, "simulation_mode", True)
-    kwargs['strategy'] = 'brute_force'
-
-    #last position argument is tune_params
-    tune_params = args[-1]
-
-    #find optimum
-    kwargs["strategy"] = "brute_force"
-    results, _ = kernel_tuner.tune_kernel(*args, **kwargs)
-    optimum = min(results, key=lambda p: p["time"])["time"]
-
-    #could throw a warning for the kwargs that will be overwritten, strategy(_options)
-    kwargs["strategy"] = target_strategy
-
-    parameter_space = itertools.product(*hyper_params.values())
-    all_results = []
-
-    for params in parameter_space:
-        strategy_options = dict(zip(hyper_params.keys(), params))
-
-        kwargs["strategy_options"] = strategy_options
-
-        fevals = []
-        p_of_opt = []
-        for _ in range(100):
-            #measure
-            with warnings.catch_warnings():
-                warnings.simplefilter("ignore")
-                results, _ = kernel_tuner.tune_kernel(*args, **kwargs)
-
-            #get unique function evaluations
-            unique_fevals = {",".join([str(v) for k, v in record.items() if k in tune_params])
-                             for record in results}
-
-            fevals.append(len(unique_fevals))
-            p_of_opt.append(min(results, key=lambda p: p["time"])["time"] / optimum * 100)
-
-        strategy_options["fevals"] = np.average(fevals)
-        strategy_options["fevals_std"] = np.std(fevals)
-
-        strategy_options["p_of_opt"] = np.average(p_of_opt)
-        strategy_options["p_of_opt_std"] = np.std(p_of_opt)
-
-        print(get_config_string(strategy_options))
-        all_results.append(strategy_options)
-
-    return all_results
+    put_if_not_present(kwargs, "verbose", True)
+    put_if_not_present(kwargs, "quiet", False)
+    kwargs['simulation_mode'] = False
+    kwargs['strategy'] = 'dual_annealing'
+    kwargs['verify'] = None
+    arguments = [target_strategy]
+
+    # execute the hyperparameter tuning
+    result, env = kernel_tuner.tune_kernel('hyperparamtuning', None, [], arguments, hyper_params, *args, lang='Hypertuner',
+                                    objective='score', objective_higher_is_better=True, iterations=iterations, **kwargs)
+    
+    # remove the temporary cachefile and return only unique results in order
+    cachefile.unlink()
+    result_unique = dict()
+    for r in result:
+        config_id = ",".join(str(r[k]) for k in hyper_params.keys())
+        if config_id not in result_unique:
+            result_unique[config_id] = r
+    return list(result_unique.values()), env
+
+if __name__ == "__main__":  # TODO remove in production
```

Comment:
What is the status of the hyperparameter tuning code present in this branch? Should this be removed?